### PR TITLE
Extensions: Simplify directories

### PR DESCRIPTION
**Pull request by roccotripaldi**
Fixes a small thread of dignity

#### Changes proposed in this Pull Request:

* Removes `extensions/setup` and puts its content in either the root of the extensions folder, or in `extensions/shared`
* Removes `extensions/utils` and puts its content in `extensions/shared`

#### Testing instructions:

- run `yarn build` and get no errors
- verify that extensions in the post editor look ok

#### Proposed changelog entry for your changes:

* None

### CODE REVIEW

File: extensions/shared/README.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+or the site admin may have filtered out the use of a certain block.
+
+## registerJetpackBlock()
+
+This util will only register a block if it meets the availability requirements described above.
+
+
+## registerJetpackPlugin()
+
+This util will only register a Gutenberg plugin if it meets the availability requirements described above.
```
Comment (simison) [inline extensions/shared/README.md]: We probably need to tell about these methods in more visible `/extensions/README.md` as part of documentation to get folks started building blocks.

File: extensions/shared/block-styles.js [hunk 2]
```diff
@@ -34,19 +34,3 @@ export function getActiveStyleName( styles, className ) {
...
-/**
- * Checks if className has a class selector starting with `is-style-`
- * Does not check validity of found style.
- *
- * @param  {String}  classNames Optional. Space separated classNames. Defaults to ''.
- * @return {Boolean} true if `classNames` has a Gutenberg style class
- */
-export function hasStyleClass( classNames = '' ) {
-	return classNames.split( ' ' ).some( className => className.startsWith( 'is-style-' ) );
-}
```
Comment (simison) [inline extensions/shared/block-styles.js]: Confirmed that these are not used anywhere; I think they used to be in Tiled gallery.

Well spotted!

File: extensions/shared/register-jetpack-block.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { registerBlockType } from '@wordpress/blocks';
 /**
  * Internal dependencies
  */
-import extensionList from '../setup/index.json';
+import extensionList from '../index.json';
```
Comment (simison) [inline extensions/shared/register-jetpack-block.js]: Nice move, like this people will find this right away when they start developing blocks.

**Review by simison — DISMISSED** [review]
Super clear and builds nicely 👍 

Tested:
```
yarn run build-extensions
yarn run test-extensions
```

**Review by kraftbj — DISMISSED** [review]
Ran the build and spot-check the post editor, so far so good.

I'm manually setting this to allow merging overriding the failed wpcom check. Since this is a big change, please merge when you have the bandwidth to make the wp.com merge happen. I'm forcing a diff to be created, but due to other Jetpack changes that didn't merge yet, we're running into conflicts.

**Comment by simison** [comment]
We were removing some test files from wpcom (they don't need to be synced) so might be that's why there was a conflict

D26665-code

**Comment by simison** [comment]
Rebased here, didn't touch D26668-code yet.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**